main.go

```diff
@@ -6,31 +6,38 @@ import (
 	"os"
 	"strings"
 
+	"github.com/bitrise-io/go-steputils/stepconf"
 	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-utils/fileutil"
 	"github.com/bitrise-io/go-utils/log"
+	"github.com/bitrise-tools/go-steputils/cache"
 	"github.com/kballard/go-shellquote"
 )
 
+const (
+	cacheInputNone = "none"
+	cacheInputAll  = "all"
+)
+
 // ConfigsModel ...
 type ConfigsModel struct {
-	Packages string
-	Options  string
-	Upgrade  string
+	Packages   string `env:"packages,required"`
+	Options    string `env:"options"`
+	Upgrade    string `env:"upgrade"`
+	CacheLevel string `env:"cache_level,opt[all,none]"`
 }
 
-func createConfigsModelFromEnvs() ConfigsModel {
-	return ConfigsModel{
-		Packages: os.Getenv("packages"),
-		Options:  os.Getenv("options"),
-		Upgrade:  os.Getenv("upgrade"),
-	}
+func fail(format string, v ...interface{}) {
+	log.Errorf(format, v...)
+	os.Exit(1)
 }
 
 func (configs ConfigsModel) print() {
 	log.Infof("Configs:")
 	log.Printf("- Packages: %s", configs.Packages)
 	log.Printf("- Options: %s", configs.Options)
 	log.Printf("- Upgrade: %s", configs.Upgrade)
+	log.Printf("- Cache Level: %s", configs.CacheLevel)
 }
 
 func (configs ConfigsModel) validate() error {
```

Comment:
please remove this deprecated method, stepconf.Parse has it's own built-in validation